ChangeLog.md

```diff
@@ -1,5 +1,12 @@
 ### 0.4.0
 
+#### Upgrade Notes
+
+* When updating from `chruby` < 0.4 to `chruby` >= 0.4.0, you will need to
+  re-install your gems because the gem directories will be different. It is also
+  recommended to clean old gem directories which will no longer be used with
+  `rm -rf ~/.gem`.
```

Comment:
Maybe a cleaner way of deleting old chruby gem dirs would be something like `rm -rf ~/.gem/{ruby,jruby,rbx}/`. Another option might be to mass-rename the directories from `~/.gem/$ruby/$version` to `~/.gem/$ruby-$version` using `xargs`.

Reply:
> Maybe a cleaner way of deleting old chruby gem dirs would be something like `rm -rf ~/.gem/{ruby,jruby,rbx}/`.

Would need to be `rm -rf ~/.gem/{ruby,jruby,rbx,truffleruby}/`.
Could change to that, but I'm not sure what that would save compared to `rm -rf ~/.gem`.
Saving `~/.gem/specs/` doesn't seem really important, it's just a cache AFAIK.

> Another option might be to mass-rename the directories from `~/.gem/$ruby/$version` to `~/.gem/$ruby-$version` using `xargs`.

That's only correct if Rubies were all installed in `$ruby_engine-$ruby_version` format and even then it's actually:
`~/.gem/$ruby_engine/$ruby_version` => `~/.gem/$ruby_engine/$ruby_engine_version`
And we probably can't easily guess the `$ruby_engine_version`, except for MRI where it's the same.

Reply:
@eregon system ruby can install into `~/.gem/` as well. For example, on Fedora they configure rubygems to install binstubs into `~/bin/` and other bits into `~/.gem/` by default for regular users:

    $ rm -rf ~/.gem/ ~/bin/*
    $ gem install bundler
    Fetching bundler-2.0.2.gem
    Successfully installed bundler-2.0.2
    Parsing documentation for bundler-2.0.2
    Installing ri documentation for bundler-2.0.2
    Done installing documentation for bundler after 2 seconds
    1 gem installed
    $ ls ~/.gem/
    ruby specs
    $ ls .gem/ruby
    build_info  cache  doc  extensions  gems  specifications
    $ ls ~/bin/
    bundle bundler

I do not want to recommend deleting all of `~/.gem/` and raise the ire of users once they realize their non-chruby gems are suddenly gone.

Reply:
Ah, one issue with selectively rm'ing the chruby generated GEM_HOMEs, `~/.gem/ruby/X.Y.Z` shadows the legitimate `~/.gem/ruby/` directory generated by system RubyGems. Perhaps something like this:

    $ ls ~/.gem/ruby
    2.6.3  build_info  cache  doc  extensions  gems  specifications
    $ ls -d ~/.gem/ruby/[0-9].[0-9].[0-9]
    /home/postmodern/.gem/ruby/2.6.3

Reply:
RE mapping `~/.gem/$RUBY_ENGINE/$RUBY_VERSION` to `~/.gem/$RUBY_ENGINE/$release_version`, we could use the same trick that chruby uses to query `RUBY_VERSION` and try invoking the absolute `#!/path/to/ruby` but with `-e 'p RUBY_ENGINE_VERSION'`.

Reply:
`~/.gem/{ruby,jruby,rbx,truffleruby}/[0-9].[0-9].[0-9]` sounds good to me, I'll use that in the upgrade notes.

I'm against documenting about moving gem directories, as that's bound to fail quite badly if anything ever stores an absolute path.